test/alternator/test_streams.py

```diff
@@ -0,0 +1,82 @@
+# Tests for stream operations: ListStreams, DescribeStream, GetShardIterator.
+
+import pytest
+from botocore.exceptions import ClientError
+from util import list_tables, test_table_name, create_test_table, random_string
+
+stream_types = [ 'OLD_IMAGE', 'NEW_IMAGE', 'KEYS_ONLY', 'NEW_AND_OLD_IMAGES']
+
+def test_list_streams(dynamodb, dynamostreams):
+    for type in stream_types:
+        table = create_test_table(dynamodb,
+            StreamSpecification={'StreamEnabled': True, 'StreamViewType': type},
+            KeySchema=[{ 'AttributeName': 'p', 'KeyType': 'HASH' }],
+            AttributeDefinitions=[{ 'AttributeName': 'p', 'AttributeType': 'S' }]);
+
+        streams = dynamostreams.list_streams(TableName=table.name)
+
+        assert streams
+        assert streams.get('LastEvaluatedStreamArn')
+        assert streams.get('Streams')
+                    
+        table.delete();
+
+def test_describe_stream(dynamodb, dynamostreams):
+    table = create_test_table(dynamodb,
+        StreamSpecification={'StreamEnabled': True, 'StreamViewType': 'KEYS_ONLY'},
+        KeySchema=[{ 'AttributeName': 'p', 'KeyType': 'HASH' }],
+        AttributeDefinitions=[{ 'AttributeName': 'p', 'AttributeType': 'S' }]);
+
+    streams = dynamostreams.list_streams(TableName=table.name)
+    arn = streams['Streams'][0]['StreamArn'];
+
+    desc = dynamostreams.describe_stream(StreamArn=arn)
+
+    print(desc)
+
+    assert desc;
+    assert desc.get('StreamDescription')
+    assert desc['StreamDescription']['StreamArn'] == arn
+    assert desc['StreamDescription']['StreamStatus'] != 'DISABLED'
+    assert desc['StreamDescription']['StreamViewType'] == 'KEYS_ONLY'
+    assert desc['StreamDescription']['TableName'] == table.name
+    assert desc['StreamDescription'].get('Shards')
+    assert desc['StreamDescription']['Shards'][0].get('ShardId')
+    assert desc['StreamDescription']['Shards'][0].get('SequenceNumberRange')
+    assert desc['StreamDescription']['Shards'][0]['SequenceNumberRange'].get('StartingSequenceNumber')
+
+def test_get_shard_iterator(dynamodb, dynamostreams):
+    table = create_test_table(dynamodb,
+        StreamSpecification={'StreamEnabled': True, 'StreamViewType': 'KEYS_ONLY'},
+        KeySchema=[{ 'AttributeName': 'p', 'KeyType': 'HASH' }],
+        AttributeDefinitions=[{ 'AttributeName': 'p', 'AttributeType': 'S' }]);
+
+    streams = dynamostreams.list_streams(TableName=table.name)
+    arn = streams['Streams'][0]['StreamArn'];
+    desc = dynamostreams.describe_stream(StreamArn=arn)
+
+    shard_id = desc['StreamDescription']['Shards'][0]['ShardId'];
+    seq = desc['StreamDescription']['Shards'][0]['SequenceNumberRange']['StartingSequenceNumber'];
+
+    for type in ['AT_SEQUENCE_NUMBER', 'AFTER_SEQUENCE_NUMBER', 'TRIM_HORIZON', 'LATEST']: 
+        iter = dynamostreams.get_shard_iterator(
+            StreamArn=arn, ShardId=shard_id, ShardIteratorType=type, SequenceNumber=seq
+        )
+        assert iter.get('ShardIterator')
```

Comment:
Please also check the error case - e.g., trying to pass an invalid type, and invalid arn, shardid, etc. You will be able to run it on both DynamoDB and Alternator and see you get the same error types, similar error messages (it doesn't need to be the same wording, but it's convenient to have one common word you can search for), etc.

Reply:
Yes. Eventually.

Reply:
I usually run all the tests on DynamoDB **first** (pytest --aws). A lot of times even before writing the implementation in Scylla. It's so convenient - it really helps you write tests which verify real expected behavior, not what you *think* is the expected behavior. Don't do this (running against DynamoDB) too "eventually" - you are risking writing a lot of code which you can suddenly realize was based on wrong assumptions (I don't think it's the case, but still...).